File: mobile/lib/features/boarding/data/models/validate_boarding_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'validate_boarding_response.g.dart';

/// Response from POST /api/boarding/validate.
/// [valid] is false when the QR is invalid, trip not active, etc.
@JsonSerializable()
class ValidateBoardingResponse {
  const ValidateBoardingResponse({
    required this.valid,
    required this.message,
    this.stopName,
    this.routeName,
    this.tripId,
    this.fare,
  });

  final bool    valid;
  final String  message;
  final String? stopName;
  final String? routeName;
  final int?    tripId;
  final double? fare;

  factory ValidateBoardingResponse.fromJson(Map<String, dynamic> json) =>
      _$ValidateBoardingResponseFromJson(json);

  Map<String, dynamic> toJson() => _$ValidateBoardingResponseToJson(this);
}

File: backend/src/main/java/com/shuttle/boarding/dto/ValidateBoardingResponse.java
package com.shuttle.boarding.dto;

import java.math.BigDecimal;

public record ValidateBoardingResponse(
        boolean    valid,
        String     message,
        String     stopName,
        String     routeName,
        Long       tripId,
        BigDecimal fare        // null when no fare is configured
) {}
